### ISSUE TITLE
x:expect href doesn't work

### ISSUE THREAD

**Issue by AirQuick** [xspec/xspec#30]
**test.xml** as a document
```xml
<?xml version="1.0" encoding="UTF-8"?>
<foo/>
```

**test.xsl** which just returns the context document
```xml
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet version="2.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
	<xsl:template as="document-node()" match="document-node()">
		<xsl:sequence select="." />
	</xsl:template>
</xsl:stylesheet>
```

**test.xspec** which tests **test.xsl** using href=**test.xml**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<x:description stylesheet="test.xsl" xmlns:x="http://www.jenitennison.com/xslt/xspec">
	<x:scenario label="Scenario-1">
		<x:context href="test.xml" />
		<x:expect href="test.xml" label="Expect-1" />
	</x:scenario>
</x:description>
```

With XSpec 0.3.0 which I guess is the last build released as stable, the test completes successfully:
```
...
Running Tests...
Testing with SAXON EE 9.6.0.10
Scenario-1
Expect-1

Formatting Report...
passed: 1 / pending: 0 / failed: 0 / total: 1
Done.
```

With the latest master, it fails:
```
Creating Test Stylesheet...
ERROR in scenario "Expect-1": unknown expect combination
    content : false
    @context: false
    @assert : false
    @select : false
    @test   : false
Error at xsl:message on line 703 of generate-xspec-tests.xsl:
  XTMM9000: Processing terminated by xsl:message at line 703 in generate-xspec-tests.xsl
...
```

Looks like the problem was reported immediately after the release of 0.4.0 RC, but neglected.
https://groups.google.com/d/msg/xspec-users/b_V-jz6WDus/tETcL_Wz3J0J

**Comment by cirulls** [comment]
Thanks for reporting this issue, I'll have a look and keep you posted. Please allow me 7-10 days to get back to you as it is Christmas time.

**Comment by cirulls** [comment]
This seems to be a regression bug introduced in v0.4.0-RC. The issue seems to be inside [generate-xspec-tests.xsl](https://github.com/xspec/xspec/blob/master/src/compiler/generate-xspec-tests.xsl). This file contains two templates named ```x:output-expect-FIXME-TOREMOVE``` and ```x:output-expect```. Looking at the log history, ```x:output-expect-FIXME-TOREMOVE``` was previously named ```x:output-expect``` in v0.3.0 and seemed to work fine. The new ```x:output-expect``` - which also introduced attributes ```context``` and ```assert``` in ```x:expect``` - was added in v0.4.0-RC in commit 70306e6df182fc358e996bd61021d6a1a4a0cc75 to fix https://github.com/expath/xspec/issues/5 with the comment 'First implementation of assert and context (issue 5), for XSLT'. 

Looking at https://github.com/expath/xspec/issues/5, it seems to me that the test provided didn't make much sense in the first place (as the reporter said) and triggered the introduction of the assert attribute to cope with boolean values. I think if the example test would have been correctly written as ``` test="boolean(self::bogus)"``` as noted in the comment, there would have been no need for the assert attribute. 

In fact, this change seems to cause several regression bugs as reported in this ticket as well as #26 and #33. Reverting to the previous version of ```x:output-expect``` (i.e. the one currently named ```x:output-expect-FIXME-TOREMOVE```) seems to fix all the reported issues. 

I could not find any documentation on the use of attributes ```assert``` and ```context``` for ```x:expect```. Looking at the test suite, all the tests using attribute ```assert``` are generating exactly the same results as for attribute ```test``` - in fact, all the tests are duplicated for both attributes ```test``` and ```assert```. This makes me assume that ```test``` and ```assert``` have basically the same functionality apart from the boolean case that can be solved by writing an XSpec test as ``` test="boolean(self::bogus)"```. 

As reported by @yamahito in #26, Oxygen uses ```x:output-expect``` from v0.3.0 (the one currently named ```x:output-expect-FIXME-TOREMOVE```). Therefore I'm inclined to revert to that version of ```x:output-expect``` as it is much more stable. This also removes the ```assert``` and ```context``` attributes of ```x:expect```, unfortunately I was not able to find any documentation and I don't know what these new attributes are supposed to achieve (```assert``` really seems a duplicate of ```test```). 

I committed branch https://github.com/xspec/xspec/tree/bugfix/30 which uses that version of ```x:output-expect``` and removes the redundant tests using the ```assert``` attribute in the test suite. This should fix in one go #26, #30 and #33 as well as making redundant #31 and #32. I added 3 additional tests to cover the above tickets, I used the examples provided by @AirQuick and @yamahito.

@AirQuick: could you check that branch https://github.com/xspec/xspec/tree/bugfix/30 fixes the issues you raised and does not introduce any other issue at your end?

**Comment by AirQuick** [comment]
Thanks for the investigation. I'll test it with my tests.

The way `x:expect` content and attributes work together is a bit confusing. I guess `assert` and `context` was introduced with very good intention to make it more straightforward. Unfortunately they should have some documentation and test...

**Comment by AirQuick** [comment]
Tested https://github.com/xspec/xspec/tree/bugfix/30 on Windows 10 + Saxon-EE 9.7.0.14 + Java 1.8 u111 x64. It worked fine with my existing test cases.
These issues are not reproduced: https://github.com/xspec/xspec/issues/30 https://github.com/xspec/xspec/issues/33
And I agree these two are not applicable anymore: https://github.com/xspec/xspec/issues/31 https://github.com/xspec/xspec/issues/32

**Comment by cirulls** [comment]
I merged #38 and close this issue.
